### DISCUSSION TITLE
Why is DHCP happening when I haven't specified DHCP?

### DISCUSSION THREAD

**Discussion by mike-somethingnooneelsehasusedyet** [ipxe/ipxe#1319] [Q&A] [answered]
I'm booting servers using a custom built iPXE iso using an embed script. E.g. the command used to make an iso for a given server looks like

`$ make -j4 bin/ipxe.iso EMBED=/tmp/qux_ipxescript`

and `/tmp/qux_ipxescript` looks like

```
#!ipxe
  
ifopen net0
set net0/ip 10.70.13.124
set net0/netmask 255.255.255.0
set net0/gateway 10.70.13.254
# ipxe only supports one dns server
set dns 10.70.12.1
# ifopen net0 - i also tried ifopen here makes no difference to behaviour
prompt -k i -t 10000 Press i within 10 seconds to do something other than boot from disk || exit
echo "Configured IP Details:"
show net0/ip
show net0/netmask
show net0/gateway
show dns
echo -n Type init and press Enter to initiate installation of OS: 
read blah
kernel http://repos-almalinux9.foo.bar/repos/test/almalinux9-kickstart/isolinux/vmlinuz ip=dhcp text inst.ks=https://repos-almalinux9.foo.bar/kickstart/qux/qux.cfg biosdevname=0 net.ifnames=0
${blah}rd http://repos-almalinux9.foo.bar/repos/test/almalinux9-kickstart/isolinux/initrd.img
boot
```

I've been doing this for a while and all the time thinking that because I'm specifying IP address etc that no DHCP is involved. I've recently discovered that when a server is booted from an iPXE iso and OS install initiated our DHCP server logs show that server making a DHCP request and being allocated an IP address. Checking in a shell while the OS installer is running shows the IP address specified in the embed script being used. E.g. If I boot a server using using the embed script shown above, I press "i", the network config for 10.70.13.124 set in the script is displayed, then I type "init" and press Enter, then DHCP logs show the server being assigned an, say, 10.70.13.50, could be something else, when I look at network config on the server while the installer is running it shows it's using 10.70.13.124.

Please can anyone explain why DHCP is happening and, ideally, how to prevent it.

**Comment by mcb30** [comment, marked as answer]
> Please can anyone explain why DHCP is happening and, ideally, how to prevent it.

You are specifying `ip=dhcp` in your kernel command line.  This will be causing your booted Linux OS to issue a DHCP request.

**Reply by mike-somethingnooneelsehasusedyet** [reply]
Thank you. I was so focussed on the iPXE stuff I totally failed to think about or look at the kernel options carried over from when installs were initiated via a different method. I considered trying to delete the question due to embarrassment but maybe one day someone else might find it and it'll make them realise they've done the same thing.

**Comment by NiKiZe** [comment]
A good tool to investigate network traffic is to use tcpdump, with -v for verbose it will even display most parts of DHCP traffic in readable text, the client identifier among other things.
